### PR TITLE
Remove vendored source code

### PR DESCRIPTION
**Pull request by ryanking13**
This PR adds more compatibility codes to drop vendored Pyodide source code.

- to_js
- get_dynlibs
- wheel_dist_info_dir

`wheel_dist_info_dir` is copied from the original Pyodide source code, while other two are dummy functions.

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import sys
+
+sys.path.append("micropip")
```
Comment (rth) [inline conftest.py:3]: Are you sure about this? Normally it shouldn't be necessary.

**Review by rth — APPROVED** [review]
Thanks @ryanking13 this is very clearly a nice improvement!

Though I think it would still make sense to release https://github.com/pyodide/pyodide/issues/1839 and depend on `pyodide-py` for tests, so we are sure we are using exactly the same version of canonicalize_name etc

File: conftest.py (hunk 1 above)
Comment (ryanking13) [reply conftest.py:3]: I'm not sure why but that was needed for me to run test locally. I'll remove that file and let's see if CI complains.

**Comment by ryanking13** [comment]
> Though I think it would still make sense to release [pyodide/pyodide#1839](https://github.com/pyodide/pyodide/issues/1839) and depend on `pyodide-py` for tests, so we are sure we are using exactly the same version of canonicalize_name etc

Yes that would be optimal.

File: conftest.py (hunk 1 above)
Comment (ryanking13) [reply conftest.py]: Okay, it works. Maybe my local setting was wrong.
(The test failure is about codecov upload)

Comment (hoodmane) [reply conftest.py]: > (The test failure is about codecov upload)

This happens a lot, it's a bit annoying. If it's going to happen regularly, maybe we can figure out a way to not fail the test when the upload fails.

**Comment by ryanking13** [comment]
@rth It seems like I don't have permission to merge this (also in `matplotlib-pyodide`).

**Comment by rth** [comment]
> It seems like I don't have permission to merge this (also in matplotlib-pyodide).

Should be fixed now. CI is failing here though.

**Comment by ryanking13** [comment]
> > It seems like I don't have permission to merge this (also in matplotlib-pyodide).
> 
> Should be fixed now. CI is failing here though.

The CI failure happens in codecov upload, so I think it is okay to merge this. I'm not sure why codecov shows a lot of 404 errors.

**Comment by rth** [comment]
Thanks!

There is a setting to not fail on upload errors as far as I remember.

**Pull request merged**